server/src/ResolveLocation.c

```diff
@@ -0,0 +1,114 @@
+/*
+ * kalman.c
+ *
+ *  Created on: 23 nov 2017
+ *      Author: alexs
+ */
+
+
+#include <math.h>
+#include <stdint.h>
+#include "../include/ResolveLocation.h"
+
+float rssi2Power(float rssi)
+{
+
+	uint32_t powerCalibrator = -45; //dBm
+	return (rssi + powerCalibrator);
+}
+
+
+float kalmanProcess(float rssiBuffer[NUMBER_SAMPLES])
+{
+	float Estimate[NUMBER_SAMPLES+1];
+	float kalmanGain = 0;
+	float initialErrorEstimate = INITIAL_ERROR_ESTIMATE;
+	//float initialMeasurement;
+
+	Estimate[0] = INITIAL_ESTIMATE;
+
+	for (uint16_t i = 0; i < NUMBER_SAMPLES ; i++)
+	{
+		kalmanGain           = initialErrorEstimate / (initialErrorEstimate + INITIAL_ERROR_MEASUREMENT);
+		Estimate[i + 1]      = Estimate[i] + (kalmanGain * (rssiBuffer[i] - Estimate[i]));
+		initialErrorEstimate = (1 - kalmanGain) * initialErrorEstimate;
+	}
+
+	return Estimate[NUMBER_SAMPLES+1];
+}
+
+
+
+float power2distance(float Power_d)
+{
+	return (distance_o * pow((POWER_do - Power_d)/(10 * n_factor),10));
+}
+
+
+uint32_t trilateration_process(cord_t* nodePosition, struct co_ordinate_distance_tag trilaterationParams[TRILATERAT_NUMBER_NODES])
```

Comment:
If we are supposed to return a `uint32_t` where is the `return` statement? :smile:

Reply:
:grin: